ui/src/data/source.tsx

```diff
@@ -1333,101 +1312,98 @@ function generateOccurrenceFilter(
-                      operator: tanagra.BinaryOperatorV2.GreaterThan,
-                      value: literalFromDataValue(min),
-                    },
-                  },
+              },
+            },
+            {
+              filterType: tanagra.FilterV2FilterTypeEnum.Attribute,
+              filterUnion: {
+                attributeFilter: {
+                  attribute: filter.attribute,
+                  operator: tanagra.BinaryOperatorV2.GreaterThan,
+                  value: literalFromDataValue(min),
                 },
-              ]
-            )
-          )
-        ),
-        entity.entity,
-      ];
+              },
+            },
+          ])
+        )
+      );
     }
     if (filter.values?.length) {
-      return [
-        makeBooleanLogicFilter(
-          tanagra.BooleanLogicFilterV2OperatorEnum.Or,
-          filter.values.map((value) => ({
-            filterType: tanagra.FilterV2FilterTypeEnum.Attribute,
-            filterUnion: {
-              attributeFilter: {
-                attribute: filter.attribute,
-                operator: tanagra.BinaryOperatorV2.Equals,
-                value: literalFromDataValue(value),
-              },
+      return makeBooleanLogicFilter(
+        tanagra.BooleanLogicFilterV2OperatorEnum.Or,
+        filter.values.map((value) => ({
+          filterType: tanagra.FilterV2FilterTypeEnum.Attribute,
+          filterUnion: {
+            attributeFilter: {
+              attribute: filter.attribute,
+              operator: tanagra.BinaryOperatorV2.Equals,
+              value: literalFromDataValue(value),
             },
-          }))
-        ),
-        entity.entity,
-      ];
+          },
+        }))
+      );
     }
 
-    return [null, ""];
+    return null;
   }
 
   if (isTextFilter(filter)) {
     if (filter.text.length === 0) {
-      return [null, ""];
+      return null;
     }
 
-    return [
-      {
-        filterType: tanagra.FilterV2FilterTypeEnum.Text,
-        filterUnion: {
-          textFilter: {
-            matchType: tanagra.TextFilterV2MatchTypeEnum.ExactMatch,
-            text: filter.text,
-            attribute: filter.attribute,
-          },
+    return {
+      filterType: tanagra.FilterV2FilterTypeEnum.Text,
+      filterUnion: {
+        textFilter: {
+          matchType: tanagra.TextFilterV2MatchTypeEnum.ExactMatch,
+          text: filter.text,
+          attribute: filter.attribute,
         },
       },
-      findEntity(filter.occurrenceID, source.config).entity,
-    ];
+    };
   }
 
   throw new Error(`Unknown filter type: ${JSON.stringify(filter)}`);
 }
 
+function arrayFilterOperator(
+  filter: ArrayFilter
+): tanagra.BooleanLogicFilterV2OperatorEnum {
+  if (!isValid(filter.operator.min) && !isValid(filter.operator.max)) {
+    return tanagra.BooleanLogicFilterV2OperatorEnum.And;
+  }
+  if (filter.operator.min === 1 && !isValid(filter.operator.max)) {
+    return tanagra.BooleanLogicFilterV2OperatorEnum.Or;
+  }
```

Comment:
Is this function kind of overloading the `filter.operator` property to represent both min/max and or/and? Not objecting, just trying to follow along. And if so, why did you choose to do that instead of add new filter operators?

Reply:
I made a generic boolean filter with min and max with the idea we could someday support things like "at least 3 out of 5 are true" and "at most 2 out of 3 are true". AND ( x out of x) and OR (at least 1 out of x) are just special cases of that.

This function also isn't new or changed. The function after it got merged into the function before and the diff made a mess of this one.